src/WebApp/Extensions/Extensions.cs

```diff
@@ -91,17 +94,22 @@ public static void AddAuthenticationServices(this IHostApplicationBuilder builde
     private static void AddAIServices(this IHostApplicationBuilder builder)
     {
         var openAIOptions = builder.Configuration.GetSection("AI").Get<AIOptions>()?.OpenAI;
-        if (!string.IsNullOrWhiteSpace(openAIOptions?.ApiKey))
+        var deploymentName = openAIOptions?.ChatModel;
+
+        if (!string.IsNullOrWhiteSpace(deploymentName))
         {
-            var kernelBuilder = builder.Services.AddKernel();
-            if (!string.IsNullOrWhiteSpace(openAIOptions.Endpoint))
-            {
-                kernelBuilder.AddAzureOpenAIChatCompletion(openAIOptions.ChatModel, openAIOptions.Endpoint, openAIOptions.ApiKey);
-            }
-            else
+            builder.Services.AddKernel();
+            builder.AddAzureOpenAI("chatOpenAi");
```

Comment:
How does the app behave when the dev hasn't configured openai? Before if it wasn't configured the Kernel wasn't added to DI, and the app checks for this. See 

https://github.com/dotnet/eShop/blob/63e82c9e75b1a50cd28499240fcbc2942bdc6939/src/WebApp/Components/Chatbot/Chatbot.razor#L56-L57

Reply:
That is the point of the `Enable` property since there is no Key property to check anymore. I just did it in one section.

How do we want it to work? `Enable` flag? Try to resolve the service?

Reply:
> but the AppHost fails if there is none defined

How does it fail? Will this get better when we change it to be "ConnectionProperties/Info" instead of "ConnectionString"?

We have an "optional" concept on WithReference. See https://github.com/dotnet/aspire/blob/360c6a9e25d43b71921584288c609032cedd9fcf/src/Aspire.Hosting/Extensions/ResourceBuilderExtensions.cs#L146

> How do we want it to work? Enable flag? Try to resolve the service?

My opinion is that we don't have an "Enable" flag, but instead detect if the `Key` is configured.